namesys/namesys.go

```diff
@@ -1,59 +1,95 @@
 package namesys
 
 import (
+	"strings"
+
 	context "github.com/ipfs/go-ipfs/Godeps/_workspace/src/golang.org/x/net/context"
 	ci "github.com/ipfs/go-ipfs/p2p/crypto"
 	path "github.com/ipfs/go-ipfs/path"
 	routing "github.com/ipfs/go-ipfs/routing"
 )
 
-// ipnsNameSystem implements IPNS naming.
+// ipfs NameSystem implements generic IPFS naming.
```

Comment:
On Sat, May 09, 2015 at 12:12:13PM -0700, W. Trevor King wrote:
>   /<protocol>/<protocol-specific-keys>
> 
> Personally, I favor the last form to mirror
> /ipfs/<ipfs-specific-keys>.

And it's good to note that what you're storing in the alternative
protocol doesn't have to be a link.  I could store *content* under
/dns/… or in the DHT (/dht/…?).  That's not the greatest idea for
those protocols, but it's how we handle public keys now (#1045), and I
see no reason to rule it out at a protocol/implementation level.  This
also makes it easy to interact with non-IPFS immutable protocols, e.g.:

  /some-other-content-addressable-system/key-into-their-space/…

I don't see a need for the protocol cross-links to distinguish between
“this is a link into the immutable IPFS space” or “this is a link into
the mutable DNS space that the IPNS resolver understands”.  All it
needs to know is “the key I was given resolves to a reference to key
$X in protocol $Y”.